src/components/sliders/draw.js

```diff
@@ -0,0 +1,532 @@
+        b: sliderOpts.height * ({top: 1, middle: 0.5}[yanchor] || 0),
+        t: sliderOpts.height * ({bottom: 1, middle: 0.5}[yanchor] || 0)
+    });
+}
+
+function drawSlider(gd, sliderGroup, sliderOpts) {
+    // These are carefully ordered for proper z-ordering:
+    sliderGroup
+        .call(drawRail, sliderOpts)
+        .call(drawLabelGroup, sliderOpts)
+        .call(drawTicks, sliderOpts)
+        .call(drawTouchRect, gd, sliderOpts)
+        .call(drawGrip, gd, sliderOpts);
+
+    // Position the rectangle:
+    Lib.setTranslate(sliderGroup, sliderOpts.lx + sliderOpts.pad.l, sliderOpts.ly + sliderOpts.pad.t);
+
+    // Every time the slider is draw from scratch, just detach and reattach the event listeners.
+    // This could perhaps be avoided.
+    removeListeners(gd, sliderGroup, sliderOpts);
+    attachListeners(gd, sliderGroup, sliderOpts);
+
+    setActive(gd, sliderGroup, sliderOpts, sliderOpts.active, true, false);
+}
+
+function removeListeners(gd, sliderGroup, sliderOpts) {
+    var listeners = sliderOpts._input.listeners;
+    var eventNames = sliderOpts._input.eventNames;
+    if(!Array.isArray(listeners) || !Array.isArray(eventNames)) return;
+    while(listeners.length) {
+        gd._removeInternalListener(eventNames.pop(), listeners.pop());
+    }
+}
+
+function attachListeners(gd, sliderGroup, sliderOpts) {
+    var listeners = sliderOpts._input.listeners = [];
+    var eventNames = sliderOpts._input.eventNames = [];
+
+    function makeListener(eventname, updatevalue) {
+        return function(data) {
+            var value = data;
+            if(updatevalue) {
+                value = Lib.nestedProperty(data, updatevalue).get();
+            }
+
+            // If it's *currently* invoking a command an event is received,
+            // then we'll ignore the event in order to avoid complicated
+            // invinite loops.
+            if(sliderOpts._invokingCommand) return;
+
+            setActiveByLabel(gd, sliderGroup, sliderOpts, value, false, true);
+        };
+    }
+
+    for(var i = 0; i < sliderOpts.updateevent.length; i++) {
+        var updateEventName = sliderOpts.updateevent[i];
+        var updatevalue = sliderOpts.updatevalue;
+
+        var updatelistener = makeListener(updateEventName, updatevalue);
+
+        gd._internalEv.on(updateEventName, updatelistener);
+
+        eventNames.push(updateEventName);
+        listeners.push(updatelistener);
+    }
+}
+
+function drawGrip(sliderGroup, gd, sliderOpts) {
+    var grip = sliderGroup.selectAll('rect.' + constants.gripRectClass)
+        .data([0]);
+
+    grip.enter().append('rect')
+        .classed(constants.gripRectClass, true)
+        .call(attachGripEvents, gd, sliderGroup, sliderOpts)
+        .style('pointer-events', 'all');
+
+    grip.attr({
+        width: constants.gripWidth,
+        height: constants.gripHeight,
+        rx: constants.gripRadius,
+        ry: constants.gripRadius,
```

Comment:
My only design comment is: maybe we could make the grip a little more rectangular i.e. more similar to how the range slider looks:

![image](https://cloud.githubusercontent.com/assets/6675409/18962010/070a1402-863d-11e6-9ce8-27a079463751.png)


... but I'm no design expert.